File: All_File/C/quicksort.c
#include <stdio.h>
void quicksort(int[],int,int);
void main()
{
	 int arr[20];
	  int size,i;
	   printf("Enter the numbe of Element\n");
	     scanf("%d",&size);
	        printf("Enter Element\n");
	           for (i = 0; i < size; i++)
	          {
              scanf("%d",&arr[i]);     
  	
	}
	       quicksort(arr,0,size-1);
	      printf("Sorted Element are:\n");
	    for (i = 0; i <size; i++)
	     {
	   printf("%d",arr[i] );
	 }

}
      void quicksort(int arr[],int low,int high)
      {
      	int pivot,i,j,temp;
      	  if (low<high)
      	  {
      	  	pivot=low;
      	  	 i=low;
      	  	  j=high;
      	  	   while(arr[i]<=arr[pivot] && i<=high)
      	  	   {
      	  	   	i++;
      	  	   }
      	  	  while(arr[j]>arr   [pivot] && j>=low)
      	  	  {
      	  	  	j--;
      	  	  }
      	  	if(i<j){
           temp=arr[i];
         arr[i]=arr[j];

       arr[j]=temp;


      	  	}
      	  }
      	  temp=arr[j];
      	  arr[j]=arr[pivot];
      	  arr[pivot]=temp;
      	  quicksort(arr,low,j-1);
      	  quicksort(arr,j+1,high);

      }

File: All_File/Java_oop_practice_for_Job_Interview/QuickSort.java
public class QuickSort {
    public static int partation(int arr[],int low,int high) {
        int pivot = arr[high];
        int i = low - 1;
        for (int j = low; j < high; j++) {
            if (arr[j] < pivot) {
                i++;
                //swap
                int temp = arr[i];
                arr[i] = arr[j];
                arr[j] = temp;

            }
        }
        // ata pivot er thik positon korar jonno
        i++;
        int temp = arr[i];
        arr[i] = pivot;
        arr[high] = temp;
        return i;
    }
    public static void quick(int arr[], int low, int high) {
        if (low < high) {
            int pivt = partation(arr, low, high);
            quick(arr, low, pivt - 1);
            quick(arr,   , high);
    }
  }
    public static void main(String[] args) {
        int arr[] = { 3, 2, 5, 6, 9, 8 };
        int n = arr.length;
        quick(arr, 0, n - 1);
        for (int i = 0; i < n; i++) {
        System.out.println(arr[i]);
    }
    }
}
